File: backend/Models/Membership.cs
namespace backend.Models;

public class Membership
{
    public int Id { get; set; }

    public MembershipType Type { get; set; }
    public PayType PayType { get; set; }
    public MembershipStatus Status { get; set; }
    public MemberRights Rights { get; set; }
    public MemberCategory Category { get; set; }
    public RenewalStatus RenewalStatus { get; set; }

    public DateTime StartDate { get; set; }
    public DateTime? EndDate { get; set; }
    public DateTime? DateLastPaid { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    // Navigation
    public ICollection<MembershipMember> MembershipMembers { get; set; } = [];
}

File: frontend/src/types/membership.ts
export interface Membership {
  id: number;
  type: string;
  payType: string;
  status: string;
  rights: string;
  category: string;
  renewalStatus: string;
  startDate: string;
  endDate: string | null;
  dateLastPaid: string | null;
  createdAt: string;
  updatedAt: string;
  members: MembershipMember[];
}

export interface MembershipMember {
  memberId: number;
  firstName: string;
  surname: string;
  role: string;
}

export interface CreateMembership {
  type: string;
  payType: string;
  status: string;
  rights: string;
  category: string;
  renewalStatus: string;
  startDate: string;
  endDate: string | null;
  dateLastPaid: string | null;
  members: CreateMembershipMember[];
}

export interface CreateMembershipMember {
  memberId: number;
  role: string;
}

export interface UpdateMembership {
  type: string;
  payType: string;
  status: string;
  rights: string;
  category: string;
  renewalStatus: string;
  startDate: string;
  endDate: string | null;
  dateLastPaid: string | null;
  members: CreateMembershipMember[];
}
